src/App/Pages/Accounts/LoginPasswordlessViewModel.cs

```diff
@@ -0,0 +1,191 @@
+                onException: ex => _logger.Exception(ex),
+                allowsMultipleExecutions: false);
+        }
+
+        public ICommand AcceptRequestCommand { get; }
+
+        public ICommand RejectRequestCommand { get; }
+
+        public string Origin { get; set; }
+
+        public string Email
+        {
+            get => _email;
+            set
+            {
+                LogInAttempByLabel = string.Format(AppResources.LogInAttemptByOn, value, Origin);
+                SetProperty(ref _email, value);
+            }
+        }
+
+        public string FingerprintPhrase
+        {
+            get => _fingerprintPhrase;
+            set
+            {
+                FingerprintPhraseFormatted = CreateFingerprintPhrase(value);
+                SetProperty(ref _fingerprintPhrase, value);
+            }
+        }
+
+        public FormattedString FingerprintPhraseFormatted
+        {
+            get => _fingerprintPhraseFormatted;
+            set => SetProperty(ref _fingerprintPhraseFormatted, value);
+        }
+
+        public string LogInAttempByLabel
+        {
+            get => _logInAttempByLabel;
+            set => SetProperty(ref _logInAttempByLabel, value);
+        }
+
+        public string DeviceType
+        {
+            get => _deviceType;
+            set => SetProperty(ref _deviceType, value);
+        }
+
+        public string IpAddress
+        {
+            get => _ipAddress;
+            set => SetProperty(ref _ipAddress, value);
+        }
+
+        public string NearLocation
+        {
+            get => _nearLocation;
+            set => SetProperty(ref _nearLocation, value);
+        }
+
+        public DateTime RequestDate
+        {
+            get => _requestDate;
+            set
+            {
+                TimeOfRequestText = CreateRequestDate();
+                SetProperty(ref _requestDate, value);
+            }
+        }
+
+        public string TimeOfRequestText
+        {
+            get => _timeOfRequest;
+            set
+            {
+                SetProperty(ref _timeOfRequest, value);
+            }
+        }
+
+        private FormattedString CreateFingerprintPhrase(string fingerprintPhrase)
+        {
+            var fingerprintList = fingerprintPhrase.Split('-').ToList();
+            var fs = new FormattedString();
+            var lastFingerprint = fingerprintList.LastOrDefault();
+
+            foreach (var fingerprint in fingerprintList)
+            {
+                fs.Spans.Add(new Span
+                {
+                    Text = fingerprint
+                });
+
+                if (fingerprint == lastFingerprint)
+                {
+                    break;
+                }
+
+                fs.Spans.Add(new Span
+                {
+                    Text = "-",
+                    TextColor = ThemeManager.GetResourceColor("DangerColor")
+                });
+            }
+
+            return fs;
+        }
```

Comment:
could `fingerprintPhrase` be `null`?
This method should have a try...catch, if an exception is raised then it'll crash the app
Also I'd extract `'-'` to a const in some place so it's not hardcoded and we can reuse it. Also if at some moment the separator is changed or can be changed by the user it'd be easier to maintain or modify.

You can also write this simpler with Linq Aggregate:
```c#
private FormattedString CreateFingerprintPhrase(string fingerprintPhrase)
{
    try
    {
        return fingerprintPhrase?.Split(new [] { SomeConsts.FingerprintPhraseSeparator }, StringSplitOptions.RemoveEmptyEntries).Aggregate(new FormattedString(), (fs, fingerprint) => 
        {
            if(fs.Spans.Any())
            {
                fs.Spans.Add(new Span
                {
                    Text = SomeConsts.FingerprintPhraseSeparator,
                    TextColor = ThemeManager.GetResourceColor("DangerColor")
                });
            }

            fs.Spans.Add(new Span
            {
                Text = fingerprint
            });
        });
    }
    catch (Exception ex)
    {
        _logger.Exception(ex);
        return null;
    }
}
```

Reply:
it was supposed to always come and have that structure but it's prettier with the linq aggregate 👍